src/Features/LanguageServer/Protocol/Extensions/ProtocolConversions.cs

```diff
@@ -649,18 +649,23 @@ public static ProjectId ProjectContextToProjectId(LSP.VSProjectContext projectCo
         }
 
         public static async Task<DocumentOptionSet> FormattingOptionsToDocumentOptionsAsync(
-            LSP.FormattingOptions options,
+            LSP.FormattingOptions? options,
             Document document,
             CancellationToken cancellationToken)
         {
             var documentOptions = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
-            // LSP doesn't currently support indent size as an option. However, except in special
-            // circumstances, indent size is usually equivalent to tab size, so we'll just set it.
-            var updatedOptions = documentOptions
-                .WithChangedOption(Formatting.FormattingOptions.UseTabs, !options.InsertSpaces)
-                .WithChangedOption(Formatting.FormattingOptions.TabSize, options.TabSize)
-                .WithChangedOption(Formatting.FormattingOptions.IndentationSize, options.TabSize);
-            return updatedOptions;
+
+            if (options != null)
+            {
+                // LSP doesn't currently support indent size as an option. However, except in special
+                // circumstances, indent size is usually equivalent to tab size, so we'll just set it.
+                documentOptions = documentOptions
+                    .WithChangedOption(Formatting.FormattingOptions.UseTabs, !options.InsertSpaces)
+                    .WithChangedOption(Formatting.FormattingOptions.TabSize, options.TabSize)
+                    .WithChangedOption(Formatting.FormattingOptions.IndentationSize, options.TabSize);
```

Comment:
I happened to be looking at similar code to this in Razor the other day, and noticed that formatting options in LSP now have 5 properties: https://microsoft.github.io/language-server-protocol/specifications/specification-3-17/#documentFormattingOptions

Are the other two unused? or do we need to update all of these places (and in razor)?

Reply:
They aren't yet defined by the VS client shape - https://devdiv.visualstudio.com/DevDiv/_git/VSLanguageServerClient?path=%2Fsrc%2Fproduct%2FProtocol%2FLanguageServer.Protocol%2FFormattingOptions.cs&_a=contents&version=GBdevelop (and aren't passed to us either).